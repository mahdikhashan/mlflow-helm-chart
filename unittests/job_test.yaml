suite: test job

templates:
  - job.yaml

tests:
  - it: should be empty when postgres or databaseMigration is not enabled
    asserts:
      - hasDocuments:
          count: 0

  - it: should have migration job when postgres and databaseMigration are enabled
    set:
      backendStore.postgres.enabled: true
      backendStore.postgres.host: postgres-service
      backendStore.postgres.port: 5432
      backendStore.postgres.database: postgres
      backendStore.databaseMigration: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Job
      - isAPIVersion:
          of: batch/v1
  
  - it: should have dbchecker init container if enabled
    set:
      backendStore:
        databaseMigration: true
        databaseConnectionCheck: true
        postgres:
          enabled: true
          host: test-host
          database: test-database
          user: test-user
          password: test-password
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Job
      - equal:
          path: spec.template.spec.initContainers[0].name
          value: dbchecker

  - it: should match snapshot of default values
    set:
      backendStore:
        databaseMigration: true
        databaseConnectionCheck: true
        postgres:
          enabled: true
          host: test-host
          database: test-database
          user: test-user
          password: test-password
    release:
      name: mlflow
      namespace: mlflow
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
