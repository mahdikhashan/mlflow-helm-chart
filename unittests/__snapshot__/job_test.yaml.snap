should match snapshot of default values with mysql:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        helm.sh/hook: post-install,post-upgrade
        helm.sh/hook-delete-policy: hook-succeeded,hook-failed
      labels:
        app: mlflow
        app.kubernetes.io/instance: mlflow
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: mlflow
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: mlflow-1.0.0
        version: 1.0.0
      name: mlflow
    spec:
      activeDeadlineSeconds: 900
      backoffLimit: 0
      template:
        metadata:
          labels:
            app: mlflow
            app.kubernetes.io/instance: mlflow
            app.kubernetes.io/name: mlflow
        spec:
          containers:
          - args:
            - /opt/mlflow/migrations.py
            command:
            - python
            envFrom:
            - configMapRef:
                name: mlflow-env-configmap
            - secretRef:
                name: mlflow-env-secret
            image: burakince/mlflow:1.0.0
            imagePullPolicy: IfNotPresent
            name: mlflow-db-migration-job
            resources: {}
            securityContext: {}
            volumeMounts:
            - mountPath: /opt/mlflow/migrations.py
              name: migrations-config
              readOnly: true
              subPath: migrations.py
          initContainers:
          - command:
            - /script/dbchecker.sh
            envFrom:
            - configMapRef:
                name: mlflow-env-configmap
            image: busybox:1.32
            imagePullPolicy: IfNotPresent
            name: dbchecker
            volumeMounts:
            - mountPath: /script/dbchecker.sh
              name: dbchecker
              subPath: dbchecker.sh
          restartPolicy: Never
          securityContext: {}
          volumes:
          - configMap:
              name: mlflow-migrations
            name: migrations-config
          - configMap:
              defaultMode: 511
              name: mlflow-dbchecker
            name: dbchecker
      ttlSecondsAfterFinished: 100
should match snapshot of default values with postgres:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        helm.sh/hook: post-install,post-upgrade
        helm.sh/hook-delete-policy: hook-succeeded,hook-failed
      labels:
        app: mlflow
        app.kubernetes.io/instance: mlflow
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: mlflow
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: mlflow-1.0.0
        version: 1.0.0
      name: mlflow
    spec:
      activeDeadlineSeconds: 900
      backoffLimit: 0
      template:
        metadata:
          labels:
            app: mlflow
            app.kubernetes.io/instance: mlflow
            app.kubernetes.io/name: mlflow
        spec:
          containers:
          - args:
            - /opt/mlflow/migrations.py
            command:
            - python
            envFrom:
            - configMapRef:
                name: mlflow-env-configmap
            - secretRef:
                name: mlflow-env-secret
            image: burakince/mlflow:1.0.0
            imagePullPolicy: IfNotPresent
            name: mlflow-db-migration-job
            resources: {}
            securityContext: {}
            volumeMounts:
            - mountPath: /opt/mlflow/migrations.py
              name: migrations-config
              readOnly: true
              subPath: migrations.py
          initContainers:
          - command:
            - /script/dbchecker.sh
            envFrom:
            - configMapRef:
                name: mlflow-env-configmap
            image: busybox:1.32
            imagePullPolicy: IfNotPresent
            name: dbchecker
            volumeMounts:
            - mountPath: /script/dbchecker.sh
              name: dbchecker
              subPath: dbchecker.sh
          restartPolicy: Never
          securityContext: {}
          volumes:
          - configMap:
              name: mlflow-migrations
            name: migrations-config
          - configMap:
              defaultMode: 511
              name: mlflow-dbchecker
            name: dbchecker
      ttlSecondsAfterFinished: 100
